internal/controller/user_controller.go

```diff
@@ -0,0 +1,137 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	kuadrav1 "github.com/Kuadrant/kuadra/api/v1"
+)
+
+// UserReconciler reconciles a User object
+type UserReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=kuadra.kuadrant.io,resources=users,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=kuadra.kuadrant.io,resources=users/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=kuadra.kuadrant.io,resources=users/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the User object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.4/pkg/reconcile
+func (r *UserReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	user := kuadrav1.User{}
+	if err := r.Get(ctx, req.NamespacedName, &user); err != nil {
+		log.Error(err, "unable to fetch User")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	awsAccount := r.createScheme(&user, req.Namespace)
+
+	if err := controllerutil.SetControllerReference(&user, awsAccount, r.Scheme); err != nil {
+		log.Error(err, "Failed to set owner reference for AwsAccount")
+		return reconcile.Result{}, err
+	}
+
+	existingAwsAccount := r.getExistingAwsAccount(ctx, awsAccount)
+	if existingAwsAccount != nil {
+		err := r.updateAwsAccount(ctx, awsAccount, existingAwsAccount)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	} else {
+		err := r.createAwsAccount(ctx, awsAccount)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *UserReconciler) createScheme(user *kuadrav1.User, namespace string) *kuadrav1.AwsAccount {
```

Comment:
Controller looks really good! Wondering should we make this function name more descriptive... Something like `createAwsAccountScheme`?